tests/tpch/conftest.py

```diff
@@ -223,19 +228,21 @@ def spark_setup(cluster, local):
             )
         spark_dashboard = parse_url(cluster._spark_dashboard)
 
-    spark._spark_dashboard: Url = spark_dashboard
-
-    # warm start
-    from pyspark.sql import Row
+    try:
+        spark._spark_dashboard: Url = spark_dashboard
 
-    df = spark.createDataFrame(
-        [
-            Row(a=1, b=2.0, c="string1"),
-        ]
-    )
-    df.show()
+        # warm start
+        from pyspark.sql import Row
 
-    yield spark
+        df = spark.createDataFrame(
+            [
+                Row(a=1, b=2.0, c="string1"),
+            ]
+        )
+        df.show()
+        yield spark
+    finally:
+        spark.stop()
```

Comment:
just good hygiene. I'm not sure any longer if it is necessary but it feels like it shouldn't hurt